### DISCUSSION TITLE
My NanoELS-build

### DISCUSSION THREAD

**Discussion by ws786873** [kachurovskiy/nanoels#25] [General]
At present, I am doing some purchasing work and printing two parts。
![webwxgetmsgimg (2)](https://user-images.githubusercontent.com/25994407/145582571-bb6c39ac-15fb-401c-bbc4-2d02cc6ecd65.jpg)
![webwxgetmsgimg (1)](https://user-images.githubusercontent.com/25994407/145582582-94f9742c-a0fc-40cf-9503-1a76046abcde.jpg)
![webwxgetmsgimg](https://user-images.githubusercontent.com/25994407/145582597-fe840b3b-c8d8-4f88-b0c4-aca9b9f6e7aa.jpg)

**Comment by kachurovskiy** [comment]
Looks good, keep us posted! What kind of plastic is that?

**Reply by ws786873** [reply]
It's PLA. Of course, ABS will be better, but I don't have ABS now. I'm considering buying finished aluminum gears.

**Comment by ws786873** [comment]
Today, I tried to install the prints. They are too small. The screw rod needs 28mm holes, while my part is only 26.5mm. Therefore, I enlarged the scale of XY axis to 104% in the print setting, and the height of Z axis remains unchanged. The encoder base is also too small. The gap left for the wire does not match the encoder I bought. I think I need to modify the model. Today, I suddenly found that the encoder is installed with double-sided tape! Is this reliable?

**Reply by ws786873** [reply]
104% is too big.....i am printing 103%...

**Reply by kachurovskiy** [reply]
If you scale the whole print then the stepper motor won't fit the bolt pattern anymore. Do you want to modify the original 3D model to fit your use case instead? https://github.com/kachurovskiy/nanoels/blob/main/h1/stepper/stepper-mount-nema23-28mm-hole.f3d

**Reply by ws786873** [reply]
I realized that scaling would lead to the wrong overall size, so I finally chose to manually grind the hole to obtain the correct size

**Comment by ws786873** [comment]
@kachurovskiy 
![屏幕截图 2021-12-11 182943](https://user-images.githubusercontent.com/25994407/145807298-3ec7ab01-612d-4d16-8ed3-296c12f729a0.png)
I installed the nano, screen and resistor for testing. The screen lights up normally, but the buttons of 0.1, 1 and 2mm seem to correspond to only 0.1, 1 and 2? Instead of providing step selection for the + - button? There are four switches about left and right. I haven't found their logic. I may understand them after installing the motor. There are some introductions in your video, but I can't understand English. Machine translation makes me unable to understand. Can you answer the question for me? By the way, I added a 2.2k resistor to the button, and the button works normally.

**Reply by kachurovskiy** [reply]
Dear friend, congrats on making it this far into the build.

Correct, top buttons are shortcuts to 0.1, 1 and 2mm. You can change those values in the code to the ones most used by you.

You can adjust the pitch in 0.01 increments using the LEFT and RIGHT buttons next to the ON/OFF button. By holding LEFT or RIGHT buttons pressed, you can adjust the pitch faster.

Let me know if you have any questions, cheers!

**Reply by ws786873** [reply]
Thank you for your answer. After I powered on the nano today, it will refresh the screen every 7.8 seconds, so it can't work normally. Another problem is that when I connect the encoder, the encoder does not turn 360 °, but the screen has shown that it has turned 360 °. The rotation angle does not match the displayed value.

**Reply by kachurovskiy** [reply]
For encoder you'd want to make sure that `ENCODER_STEPS` is set correctly in https://github.com/kachurovskiy/nanoels/blob/main/h1/NanoEls.ino#L6 - you might need to change that value and re-upload the sketch to Arduino.

For screen refresh problems it's hard to tell what's going on, especially since you have a custom PCB, but if you could attach a video showing the problem then hopefully we could help. Cheers!

**Reply by ws786873** [reply]
Maybe I should try writing 400 or 200 in the firmware. My encoder is 600P (maybe the seller deceived me). The only difference from your encoder should be the output mode. Your encoder seems to be NPN output, but my seller just vaguely told me that it uses 5V output. There is another possible error. I didn't mention that I didn't weld when testing the encoder. I just inserted them into the hole. Maybe this will also lead to an error?

**Comment by ws786873** [comment]
I drew a new encoder base, which made some adjustments to the original size and added five 3mm holes for people who need to be fixed with screws. This is a suitable modification for me. Please make sure whether the size is appropriate before use。
[encoder base v3.zip](https://github.com/kachurovskiy/nanoels/files/7711646/encoder.base.v3.zip)It contains F3D and STL, which you can easily modify.
.

**Comment by ws786873** [comment]
@kachurovskiy I have completed the motor test. I set it according to the T60 driver. Its options are exactly the same as the CL57 driver I bought. But I have a problem with gears. The gear model you provided does not match my screw rod, so can you provide your F3D file?  I need to modify it myself

**Reply by kachurovskiy** [reply]
@ws786873 you can easily generate standard gears using a built-in script in Fusion 360 - https://youtu.be/B8A_11o7QZ0?t=60

If you're looking for HTD 5M pulleys, I've added the file in https://github.com/kachurovskiy/nanoels/blob/main/h1/stepper/htd5m.f3d - thanks!

**Reply by ws786873** [reply]
Thank you. I can continue my work

**Comment by ws786873** [comment]
![捕获](https://user-images.githubusercontent.com/25994407/146938285-3843ea04-0956-4628-8d25-6dbe8932c2df.PNG)
![捕获2](https://user-images.githubusercontent.com/25994407/146938289-5f38f985-6b50-4821-88fa-332fa228cabb.PNG)
I made some modifications to the gears to suit my lathe

**Comment by ws786873** [comment]
https://www.bilibili.com/video/BV1MQ4y1Y73x?spm_id_from=333.999.0.0
This is my motor test video

**Reply by kachurovskiy** [reply]
Nice to see it working 😀

**Comment by ws786873** [comment]
@kachurovskiy Merry Christmas! I finished fixing the encoder. I drilled and tapped on the lathe and used m3 screws to fix the encoder base. But after completion, there are some gaps between the gears. And when rotating, the encoder gear is slightly eccentric, which will cause the loss of accuracy in the subsequent operation? I will show the problem in the following video to prevent unclear description caused by automatic translation.
https://www.bilibili.com/video/BV15D4y1c77c?spm_id_from=333.999.0.0

**Reply by kachurovskiy** [reply]
Merry Christmas 🎄 The distance between the gears could be a bit lower but it's ok like this too. 3D printed gear will always be a little off center, I have a similar situation, didn't cause any problems so far. There should not be any accumulating error since the number of teeth is constant :) Cheers!

**Reply by ws786873** [reply]
OK, I should be able to finish it soon. By the way, can I write and publish articles with your project?

**Reply by kachurovskiy** [reply]
Sure, it's MIT license - https://github.com/kachurovskiy/nanoels/blob/main/LICENSE - it's now your project too 😀

**Reply by mockendon** [reply]
I printed the encoder gear with a smaller hole then drilled it on the lathe - this gave me much better concentricity.

**Reply by mockendon** [reply]
> Sure, it's MIT license - https://github.com/kachurovskiy/nanoels/blob/main/LICENSE - it's now your project too 😀

Open source at it’s best Max, and a big part of why we are all here and contributing in our own ways.  Hope you feel the kudos, its well deserved!

**Reply by ws786873** [reply]
> 当然，这是 MIT 许可证 - https://github.com/kachurovskiy/nanoels/blob/main/LICENSE - 它现在也是你的项目😀
wow~

**Reply by ws786873** [reply]
> > 当然，这是 MIT 许可证 - https://github.com/kachurovskiy/nanoels/blob/main/LICENSE - 它现在也是你的项目😀
> 
> 开源是最好的 Max，这也是为什么我们都在这里并以自己的方式做出贡献的很大一部分。希望你能感受到荣誉，这是当之无愧的！

got it~

**Comment by ws786873** [comment]
@kachurovskiy Happy New Year ~ I have finished all the assembly. Thank you again for your project. But I have some small problems. The actual movement is 1 / 2 of the input I measured my screw rod. Its thread is 1mm wide and the thread spacing is 1mm, so I set "100" in the software. Is my understanding wrong? How should I measure and set it correctly?

**Reply by kachurovskiy** [reply]
@ws786873 press LEFT and RIGHT button together to see the spindle rotation angle on the screen, then rotate the chuck by hand to make sure it's showing sensible values (changes by 360 for a full rotation). Then check MOTOR_STEPS which is 200 is you've set full steps on your driver, 4000 if you've set half-steps, etc. And finally, if none of the above helps, change LEAD_SCREW_HMM until the lead screw starts moving by the right amount.

**Comment by ws786873** [comment]
The encoder rotates one circle, indicating that it has moved 1.98mm，The actual movement is about 1mm
![115138763-55999380-a02e-11eb-9380-a12b1338584d](https://user-images.githubusercontent.com/25994407/147909042-855f95c0-a81a-4221-81d7-80cc20951d9d.png)
My driver has exactly the same parameters as T60, and I set them in the same way.
 MOTOR_STEPS  is 200，and i don't know half-steps‘meaning

**Reply by kachurovskiy** [reply]
Check the position of SW1, SW2, SW3 and SW4 switches on your driver to find the pulse/rev value in that table. Use it instead of 200 for MOTOR_STEPS. Looks like you need to put 400 there.

**Reply by ws786873** [reply]
I set 800, but I used 200 in the software. My lead screw is actually 2mm. When I set 0.5mm, it works normally. Therefore, the theoretical error is 4X. I think the problem should be solved.

**Reply by ws786873** [reply]
The problem remains unsolved。It's setting is off,on,on,on,So I set up motor_ Steps 800, but after this setting, the motor makes a sound but does not rotate.

**Reply by ws786873** [reply]
The complete setup of my driver is off,on,on,on,off,off,off,on

**Reply by ws786873** [reply]
> Check the position of SW1, SW2, SW3 and SW4 switches on your driver to find the pulse/rev value in that table. Use it instead of 200 for MOTOR_STEPS. Looks like you need to put 400 there.

By the way, I haven't found any documents about the use, such as how to adjust the feed rate, how to set the linkage feed rate when the spindle rotates, and so on

**Reply by kachurovskiy** [reply]
@ws786873 it's easy to misinterpret the position of those small switches (which side is "on"), please check whether you've truly set the driver to "off,on,on,on,off,off,off,on" and not to "on,off,off,off,on,on,on,off".

You'd also want to set SW8 to SERVO so that you don't have missed steps.

Please post the pictures of [1] your driver and [2] the position of the micro-switches set on the driver.

> adjust the feed rate, how to set the linkage feed rate when the spindle rotates

You can control the feed rate (pitch) using + and - buttons on the NanoEls - https://github.com/kachurovskiy/nanoels/#turning - as well as shortcut buttons in the top row.

Cheers!

**Reply by ws786873** [reply]
> @ws786873很容易误解那些小开关的位置（哪一侧是“on”），请检查您是否真的将驱动程序设置为“off,on,on,on,off,off,off,on”和不要“开、关、关、关、开、开、开、关”。
> 
> 您还需要将 SW8 设置为 SERVO，这样您就不会错过步骤。
> 
> 请张贴 [1] 您的驱动程序的图片和 [2] 设置在驱动程序上的微动开关的位置。
> 
> > 调整进给速度，主轴旋转时如何设置联动进给速度
> 
> 您可以使用 NanoEls 上的 + 和 - 按钮控制进给速率（间距） - https://github.com/kachurovskiy/nanoels/#turning - 以及顶行的快捷按钮。
> 
> 干杯!

@kachurovskiy   picture and video
<img width="326" alt="111" src="https://user-images.githubusercontent.com/25994407/150679866-972c9058-825b-436f-bb9d-df60d4c738c6.png">
<img width="286" alt="22" src="https://user-images.githubusercontent.com/25994407/150679865-c657a682-0faa-452d-8bd8-eb01adede8e2.png">

video:https://www.bilibili.com/video/BV13T4y1y7Hc/

**Reply by kachurovskiy** [reply]
You'd want to set SW8 to OFF (servo) so that you don't have missed steps and try using NanoEls again.

**Reply by ws786873** [reply]
> 您希望将 SW8 设置为 OFF（伺服），这样您就不会错过步骤并再次尝试使用 NanoEls。

OK,i will try tomorrow

**Reply by ws786873** [reply]
> 您希望将 SW8 设置为 OFF（伺服），这样您就不会错过步骤并再次尝试使用 NanoEls。

Everything working well now,thank you for your help again!
